libraries/AP_RangeFinder/AP_RangeFinder_RDS02UF.cpp

```diff
@@ -0,0 +1,222 @@
+            decode_state = STATE5_FC_CODE_H;
+            break;
+        case STATE5_FC_CODE_H: // fc_code high
+            rev_buffer[parserbuf_index] = data;
+            parserbuf_index++;
+            decode_state = STATE6_LENGTH_L;
+            break;
+        case STATE6_LENGTH_L: // lengh_low
+            rev_buffer[parserbuf_index] = data;
+            parserbuf_index++;
+            decode_state = STATE7_LENGTH_H;
+            break;
+        case STATE7_LENGTH_H: // lengh_high
+        {
+            uint8_t read_len = data << 8 | rev_buffer[parserbuf_index-1];
+            if ( read_len == RDS02UF_DATA_LEN)	// rds02uf data length is 10
+            {
+                rev_buffer[parserbuf_index] = data;
+                parserbuf_index++;
+                decode_state = STATE8_REAL_DATA;
+            }else{
+                parserbuf_index = 0;
+                decode_state = STATE0_SYNC_1;
+            }
+            break;
+        }
+        case STATE8_REAL_DATA: // real_data
+            rev_buffer[parserbuf_index] = data;
+            parserbuf_index++;
+            if ( parserbuf_index == (RDS02UF_HEAD_LEN + RDS02UF_PRE_DATA_LEN + RDS02UF_DATA_LEN) ) {
+                decode_state = STATE9_CRC;
+            }
+            break;
+        case STATE9_CRC: // crc
+        {
+	        uint8_t crc_data = 0;
+            crc_data = crc8(&rev_buffer[2], RDS02UF_PRE_DATA_LEN + RDS02UF_DATA_LEN);
+            rev_buffer[parserbuf_index] = data;
+            if (crc_data == data || data == 0xff) {
+                parserbuf_index++;
+                decode_state = STATE10_END_1;
+            } else {
+                parserbuf_index = 0;
+                decode_state = STATE0_SYNC_1;
+            }
+            break;
+        }
+        case STATE10_END_1: //
+            if (data == RDS02_END) {
+                rev_buffer[parserbuf_index] = data;
+                parserbuf_index++;
+                decode_state = STATE11_END_2;
+            } else {
+                parserbuf_index = 0;
+                decode_state = STATE0_SYNC_1;
+            }
+            break;
+        case STATE11_END_2: //
+        {
+            uint16_t fc_code = (rev_buffer[STATE5_FC_CODE_H] << 8 | rev_buffer[STATE4_FC_CODE_L]);
+            uint8_t err_code = rev_buffer[STATE3_ERROR_CODE];
+            if (data == RDS02_END)
+            {
+                if (fc_code == 0x03ff && err_code == 0) {// get targer information
+                    uint16_t data_fc = (rev_buffer[RDS02_DATA_FC_H] << 8 | rev_buffer[RDS02_DATA_FC_L]);
+                    if((data_fc & 0xf0f) == RDS02_TARGET_INFO){	// data_fc = 0x70C + ID * 0x10, ID: 0~0xF
+                        float distance = (rev_buffer[RDS02_DATA_Y_INDEX_H] * 256 + rev_buffer[RDS02_DATA_Y_INDEX_L]) / 100.0f;
+                        _distance_m = distance;
+                        _distance_m = MIN(MAX(RDS02UF_DIST_MIN_CM, uint16_t(_distance_m*100)), RDS02UF_DIST_MAX_CM) * 0.01f;
```

Comment:
This shouldn't be constrained - this should be used to set the rangefinder as out-of-range-low or out-of-range-high